### ISSUE TITLE
Fix "no suitable peers available" in tests

### ISSUE THREAD

**Issue by tiabc** [status-im/status-go#344]
# Problem

A lot of our tests fail with "no suitable peers available" which refers to go-ethereum and we experience constant failures which don't say anything about stability of our code.

We need some workaround or something.

Check out some failing build in Travis: https://travis-ci.org/status-im/status-go/builds

# Acceptance Criteria

Tests pass 10 out of 10 times without any "no suitable peers available" error message.

# Notes

This issue is part of #220.

# Future Steps

Work further on #220.

**Comment by tiabc** [comment]
Increasing priority as failing tests have been hindering efficient development.

**Comment by themue** [comment]
Looks like `backend.StopNode()` isn't called every time or called directly via `defer` without waiting for the notification. Changing it step by step, failures are almost gone.

Helper narrowing it down:

```
$ make ci | grep -e "=== RUN" -e "suitable peers" -e "--- FAIL"
```

**Comment by tiabc** [comment]
Great job! Thanks for the update.

**Comment by adambabik** [comment]
I can describe a situation when I could reproduce this issue. I was running an e2e test which required a node running and was sending some ether between accounts. It turned out that LES initial startup was taking a bit longer than usual and, I suppose, performing some operations on blockchain required the current highest block number. 

I added a simple check:
```go
for {
	isSyncing := les.Downloader().Synchronising()
	progress := les.Downloader().Progress()

	if !isSyncing && progress.HighestBlock > 0 && progress.CurrentBlock >= progress.HighestBlock {
		break
	}

	time.Sleep(time.Second * 10)
}
```
and the test started passing. In my case, `progress` was reporting valid `CurrentBlock` but `HighestBlock` was 0 until LES found a peer that provided this information.

**Comment by themue** [comment]
Yes, added it as `EnsureSynchronization()` to the suite and currently adding it in the tests. Looks good.

**Comment by tiabc** [comment]
@themue I hope you've added it from https://github.com/status-im/status-go/pull/271

@adambabik hm, strange use case. Anyway, even in the PR above these errors with "no suitable peers available" are present.

**Comment by themue** [comment]
@tiabc Didn't know about the one there, will take a look. Mine is base on @adambabik hint but with timeout.

**Comment by tiabc** [comment]
That issue will probably be merged today.

**Comment by themue** [comment]
Ah, will merge and compare then. Mine is in `BackendTestSuite` which is embedded.

**Comment by tiabc** [comment]
@themue sorry, what will you merge? #271? Don't do it, I'll throw away some stuff from there before merging.

**Comment by themue** [comment]
I meant I'll merge the new develop into my branch once you've merged #271.
